sonar-plugin-api/src/main/java/org/sonar/api/batch/SonarIndex.java

```diff
@@ -140,15 +136,9 @@ public final void addViolation(Violation violation) {
    */
   public abstract Measure addMeasure(Resource resource, Measure measure);
 
+  /**
+   * @deprecated since 5.2 No more design features. No op.
+   */
+  @Deprecated
   public abstract Dependency addDependency(Dependency dependency);
```

Comment:
:new: ![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Do not forget to remove this deprecated code someday.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)